test/php/gutenberg-coding-standards/Gutenberg/Sniffs/Commenting/FunctionCommentSinceTagSniff.php

```diff
@@ -48,90 +91,444 @@ public function process( File $phpcsFile, $stackPtr ) {
+		if ( empty( $version_tags ) ) {
+			$phpcs_file->addError( $missing_since_tag_error_message, $stack_pointer, $violation_code );
+			return;
+		}
+
+		foreach ( $version_tags as $since_tag_token => $version_value_token ) {
+			if ( null === $version_value_token ) {
+				$phpcs_file->addError( $missing_since_tag_error_message, $since_tag_token, $violation_code );
+				continue;
+			}
+
+			$version_value = $tokens[ $version_value_token ]['content'];
+
+			if ( static::validate_version( $version_value ) ) {
+				continue;
+			}
+
+			$phpcs_file->addError(
+				'Invalid @since version value for the "%s()" %s: "%s". Version value must be greater than or equal to 0.0.1.',
+				$version_value_token,
+				$violation_code,
+				array(
+					$function_name,
+					$is_oo_method ? 'method' : 'function',
+					$version_value,
+				)
+			);
+		}
+	}
+
+	/**
+	 * Validates the version value.
+	 *
+	 * @param string $version The version value being checked.
+	 * @return bool True if the version value is valid.
+	 */
+	protected static function validate_version( $version ) {
+		$matches = array();
+		if ( 1 === preg_match( '/^MU \((?<version>.+)\)/', $version, $matches ) ) {
+			$version = $matches['version'];
+		}
+
+		return version_compare( $version, '0.0.1', '>=' );
+	}
+
+	/**
+	 * Finds the docblock associated with a hook, starting from a specified position in the token stack.
+	 * Since a line containing a hook can include any type of tokens, this method backtracks through the tokens
+	 * to locate the first token on the current line. This token is then used as the starting point for searching the docblock.
+	 *
+	 * @param File $phpcs_file    The file being scanned.
+	 * @param int  $stack_pointer The position to start looking for the docblock.
+	 * @return array|false The start and end tokens of the docblock, or false if not found.
+	 */
+	protected static function find_hook_docblock( File $phpcs_file, $stack_pointer ) {
+		$tokens       = $phpcs_file->getTokens();
+		$current_line = $tokens[ $stack_pointer ]['line'];
+
+		for ( $i = $stack_pointer; $i >= 0; $i-- ) {
+			if ( $tokens[ $i ]['line'] < $current_line ) {
+				// The previous token is on the previous line, so the current token is the first on the line.
+				return static::find_docblock( $phpcs_file, $i + 1 );
+			}
+		}
+
+		return static::find_docblock( $phpcs_file, 0 );
+	}
+
+	/**
+	 * Finds the docblock preceding a specified position (stack pointer) in a given PHP file.
+	 * The implementation was copied from FunctionCommentSniff::process().
+	 *
+	 * @param File $phpcs_file    The file being scanned.
+	 * @param int  $stack_pointer The position (stack pointer) in the token stack from which to start searching backwards.
+	 * @return array|false An array with the starting and ending token positions of the found docblock, or false if no docblock is found.
+	 */
+	protected static function find_docblock( File $phpcs_file, $stack_pointer ) {
```

Comment:
Outside of the scope of this PR, but I wonder if it would make sense to start a conversation about adding `find_docblock()` to PHPCSUtils?